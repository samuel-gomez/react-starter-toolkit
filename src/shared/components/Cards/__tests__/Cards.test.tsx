import { render, waitFor } from '@testing-library/react';
import Cards from '../index';

const defaultProps = {
  items: [
    { id: 'id1', title: 'title1', picture: { name: 'accordion.svg', alt: 'alt_accordion' } },
    { id: 'id2', title: 'title2', picture: { name: 'accordion.svg', alt: 'alt_accordion' } },
  ],
};

describe('<Cards/>', () => {
  it('Should render Cards', async () => {
    const { asFragment, getAllByAltText } = render(<Cards {...defaultProps} />);

    await waitFor(() => expect(getAllByAltText('alt_accordion').length).toEqual(2));
    expect(asFragment()).toMatchSnapshot();
  });
});
