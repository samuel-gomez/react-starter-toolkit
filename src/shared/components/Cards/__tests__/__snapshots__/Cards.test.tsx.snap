// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Cards/> Should render Cards 1`] = `
<DocumentFragment>
  <div
    class="af-cards"
  >
    <article
      class="af-card af-skeleton"
    >
      <img
        alt="alt_accordion"
        class="af-card__image"
        data-nimg="fill"
        decoding="async"
        loading="lazy"
        sizes="(max-width: 768px) 100vw"
        src="/_next/image?url=%2Fimg.jpg&w=3840&q=75"
        srcset="/_next/image?url=%2Fimg.jpg&w=640&q=75 640w, /_next/image?url=%2Fimg.jpg&w=750&q=75 750w, /_next/image?url=%2Fimg.jpg&w=828&q=75 828w, /_next/image?url=%2Fimg.jpg&w=1080&q=75 1080w, /_next/image?url=%2Fimg.jpg&w=1200&q=75 1200w, /_next/image?url=%2Fimg.jpg&w=1920&q=75 1920w, /_next/image?url=%2Fimg.jpg&w=2048&q=75 2048w, /_next/image?url=%2Fimg.jpg&w=3840&q=75 3840w"
        style="position: absolute; height: 100%; width: 100%; left: 0px; top: 0px; right: 0px; bottom: 0px; color: transparent;"
      />
      <header
        class="af-card__header"
      >
        <h3
          class="af-card__title"
        >
          title1
        </h3>
      </header>
      <section
        class="af-card__content"
      >
        <a
          class="btn af-btn--circle af-btn--svg"
          href="https://axaguildev.github.io/design-system/"
          rel="noopener noreferrer"
          target="_blank"
          title=""
        >
          <svg
            aria-labelledby="title"
            class="af-icon-svg"
            role="img"
            viewBox="0 0 342.988 280"
          >
            <title>
              slash
            </title>
            <path
              d="M233.869 0h109.119L109.119 280H0L233.869 0zM299.988 144.545V239h-78.797l-34.203 41h156V93l-43 51.545zM43 41h78.797L156 0H0v187l43-51.545V41z"
            />
          </svg>
        </a>
        <a
          class="btn af-btn--circle af-btn--storybook"
          href="https://axaguildev.github.io/react-toolkit/latest/storybook/?path=/story/"
          rel="noopener noreferrer"
          target="_blank"
          title=""
        >
          <svg
            aria-labelledby="title"
            class="af-icon-svg"
            role="img"
            viewBox="0 0 9.6 12"
          >
            <title>
              storybook
            </title>
            <path
              d="M9.6.6C9.6.2 9.3 0 9 0h-.7v1.4s0 .1-.1.1h-.1l-.3-.4-.5.4h-.1v-.1L7.3 0 .6.5c-.3 0-.6.3-.5.6L.4 11c0 .3.3.6.6.6l8 .4c.3 0 .6-.3.6-.6V.6zM5.1 9.7c-1.4 0-2.7-.6-2.5-2.6.1-.2 1.9-.2 1.9 0 0 .8.2 1.1.6 1.1.4 0 .5-.2.5-.5 0-1.2-3-1.3-3-3.5C2.6 2.9 3.5 2 5 2c1.6 0 2.4.8 2.4 2.4-.2.2-1.8.3-1.8 0 0-.9-.4-.9-.6-.9-.2 0-.6.1-.6.5 0 1.2 3 1.1 3 3.5.1 1.4-.9 2.2-2.3 2.2z"
            />
          </svg>
        </a>
      </section>
    </article>
    <article
      class="af-card af-skeleton"
    >
      <img
        alt="alt_accordion"
        class="af-card__image"
        data-nimg="fill"
        decoding="async"
        loading="lazy"
        sizes="(max-width: 768px) 100vw"
        src="/_next/image?url=%2Fimg.jpg&w=3840&q=75"
        srcset="/_next/image?url=%2Fimg.jpg&w=640&q=75 640w, /_next/image?url=%2Fimg.jpg&w=750&q=75 750w, /_next/image?url=%2Fimg.jpg&w=828&q=75 828w, /_next/image?url=%2Fimg.jpg&w=1080&q=75 1080w, /_next/image?url=%2Fimg.jpg&w=1200&q=75 1200w, /_next/image?url=%2Fimg.jpg&w=1920&q=75 1920w, /_next/image?url=%2Fimg.jpg&w=2048&q=75 2048w, /_next/image?url=%2Fimg.jpg&w=3840&q=75 3840w"
        style="position: absolute; height: 100%; width: 100%; left: 0px; top: 0px; right: 0px; bottom: 0px; color: transparent;"
      />
      <header
        class="af-card__header"
      >
        <h3
          class="af-card__title"
        >
          title2
        </h3>
      </header>
      <section
        class="af-card__content"
      >
        <a
          class="btn af-btn--circle af-btn--svg"
          href="https://axaguildev.github.io/design-system/"
          rel="noopener noreferrer"
          target="_blank"
          title=""
        >
          <svg
            aria-labelledby="title"
            class="af-icon-svg"
            role="img"
            viewBox="0 0 342.988 280"
          >
            <title>
              slash
            </title>
            <path
              d="M233.869 0h109.119L109.119 280H0L233.869 0zM299.988 144.545V239h-78.797l-34.203 41h156V93l-43 51.545zM43 41h78.797L156 0H0v187l43-51.545V41z"
            />
          </svg>
        </a>
        <a
          class="btn af-btn--circle af-btn--storybook"
          href="https://axaguildev.github.io/react-toolkit/latest/storybook/?path=/story/"
          rel="noopener noreferrer"
          target="_blank"
          title=""
        >
          <svg
            aria-labelledby="title"
            class="af-icon-svg"
            role="img"
            viewBox="0 0 9.6 12"
          >
            <title>
              storybook
            </title>
            <path
              d="M9.6.6C9.6.2 9.3 0 9 0h-.7v1.4s0 .1-.1.1h-.1l-.3-.4-.5.4h-.1v-.1L7.3 0 .6.5c-.3 0-.6.3-.5.6L.4 11c0 .3.3.6.6.6l8 .4c.3 0 .6-.3.6-.6V.6zM5.1 9.7c-1.4 0-2.7-.6-2.5-2.6.1-.2 1.9-.2 1.9 0 0 .8.2 1.1.6 1.1.4 0 .5-.2.5-.5 0-1.2-3-1.3-3-3.5C2.6 2.9 3.5 2 5 2c1.6 0 2.4.8 2.4 2.4-.2.2-1.8.3-1.8 0 0-.9-.4-.9-.6-.9-.2 0-.6.1-.6.5 0 1.2 3 1.1 3 3.5.1 1.4-.9 2.2-2.3 2.2z"
            />
          </svg>
        </a>
      </section>
    </article>
  </div>
</DocumentFragment>
`;
